### ISSUE TITLE
使用SPU进行逻辑回归教程疑问

### ISSUE THREAD

**Issue by integrationex01** [secretflow/secretflow#791]
在学习该文档中有一个疑问：
该文档对于损失函数的实现为：
`
import jax.numpy as jnp
def sigmoid(x):
    return 1 / (1 + jnp.exp(-x))
# Outputs probability of a label being true.
def predict(W, b, inputs):
    return sigmoid(jnp.dot(inputs, W) + b)
# Training loss is the negative log-likelihood of the training examples.
def loss(W, b, inputs, targets):
    preds = predict(W, b, inputs)
    label_probs = preds * targets + (1 - preds) * (1 - targets)
    return -jnp.mean(jnp.log(label_probs))
`
其中 label_probs = preds * targets + (1 - preds) * (1 - targets)
但实际上对数似然函数的形式为： ln(preds)*targets+ln(1-preds)*(1-targets)
损失函数即平均对数似然损失的实现为：-jnp.mean(jnp.log(jnp.log(preds)*targets+jnp.log(1-preds)*(1-targets)))

由于我也是初学，各方面还请多多包涵

**Comment by deadlywing** [comment]
Hi, 确实，这里loss的表达式有误，感谢指正，我们会立刻修正哈～

Thanks

**Comment by integrationex01** [comment]
@deadlywing 可否解释一下原因呢，是因为在（0，1）范围内x与lnx数值误差不大吗？

**Comment by deadlywing** [comment]
不是哈，是因为对01二分类而言，y只能取到0或者1，那么对两种表达式而言，当y为1，结果都为-log(pred)。y为0时也是同理。所以两种表达式在这种情况下等价。

Thanks

**Comment by integrationex01** [comment]
@deadlywing 好的，了解，感谢解答
